snap/kube-controller-manager.yaml

```diff
@@ -33,4 +33,21 @@ parts:
       cp $KUBE_SNAP_ROOT/kube-controller-manager/kube-controller-manager-wrapper .
       $KUBE_SNAP_ROOT/shared/generate-configure-hook kube-controller-manager
 
-      snapcraftctl build
\ No newline at end of file
+      # By mistake --service-account-private-key-file was marked as deprecated
+      # https://github.com/kubernetes/kubernetes/pull/62722
+      # Should be fixed soon by https://github.com/kubernetes/kubernetes/pull/60270/files#diff-fc9db90cc6b68d7c3ca838a8512447e9R61
```

Comment:
This isn't how I read those PRs. It looks to me like the flag was marked deprecated as far back as 1.8, with this PR: https://github.com/kubernetes/kubernetes/pull/50289

1.8 release note:
> The `kube-cloud-controller-manager` flag `--service-account-private-key-file` was non-functional and is now deprecated.

60270 was already merged, but I think it's just a big refactor that moved stuff around. Still looking into it.

62722 has not been merged yet, but it will completely remove the --service-account-private-key-file flag.

Is the correct fix here to get the charm to stop setting it?

Reply:
Oh, nevermind. I see now. Those are cloud-controller-manager not kube-controller-manager.

/me looks again

Reply:
Yep, I can't make sense of what they're doing. Looks like the deprecation on kube-controller-manager was by accident, though, when they merged the backend options code for kube-controller-manager and kube-cloud-controller-manager.

LGTM, sorry for the false alarm :+1: